src/NUnitFramework/framework/Compatibility/ReflectionExtensions.cs

```diff
@@ -415,72 +417,36 @@ public static MethodInfo GetGetMethod(this PropertyInfo pinfo, bool nonPublic)
 
             return pinfo.GetMethod;
         }
+#endif
 
         /// <summary>
         /// Returns an array of custom attributes of the specified type applied to this member
         /// </summary>
         public static IEnumerable<T> GetAttributes<T>(this MemberInfo info, bool inherit) where T : class
         {
-            return GetAttributesImpl<T>(info.GetCustomAttributes(inherit));
+            return info.GetCustomAttributes(inherit).OfType<T>();
         }
 
         /// <summary>
         /// Returns an array of custom attributes of the specified type applied to this parameter
         /// </summary>
         public static IEnumerable<T> GetAttributes<T>(this ParameterInfo info, bool inherit) where T : class
         {
-            return GetAttributesImpl<T>(info.GetCustomAttributes(inherit));
+            return info.GetCustomAttributes(inherit).OfType<T>();
         }
 
         /// <summary>
         /// Returns an array of custom attributes of the specified type applied to this assembly
         /// </summary>
         public static IEnumerable<T> GetAttributes<T>(this Assembly asm) where T : class
         {
-            return GetAttributesImpl<T>(asm.GetCustomAttributes());
-        }
-
-        private static IEnumerable<T> GetAttributesImpl<T>(IEnumerable<Attribute> attributes) where T : class
-        {
-            var attrs = new List<T>();
-
-            attributes.Where(a => typeof(T).IsAssignableFrom(a.GetType()))
-                .All(a => { attrs.Add(a as T); return true; });
-
-            return attrs;
-        }
-    }
-
-    /// <summary>
-    /// Extensions for Assembly that are not available in .NET Standard
-    /// </summary>
-    public static class AssemblyExtensions
-    {
-        /// <summary>
-        /// DNX does not have a version of GetCustomAttributes on Assembly that takes an inherit
-        /// parameter since it doesn't make sense on Assemblies. This version just ignores the
-        /// inherit parameter.
-        /// </summary>
-        /// <param name="asm">The assembly</param>
-        /// <param name="attributeType">The type of attribute you are looking for</param>
-        /// <param name="inherit">Ignored</param>
-        /// <returns></returns>
-        public static object[] GetCustomAttributes(this Assembly asm, Type attributeType, bool inherit)
-        {
-            return asm.GetCustomAttributes(attributeType).ToArray();
```

Comment:
This had also been incorrectly returning an `Attribute[]` instance, whereas the BCL method it's polyfilling returns an `object[]` instance.